spring-integration-core/src/test/java/org/springframework/integration/handler/DelayHandlerTests.java

```diff
@@ -459,6 +468,37 @@ public void testRescheduleAndHandleAtTheSameTime() throws Exception {
 		assertNull(message);
 	}
 
+	@Test
+	public void testRescheduleForTheDateDelay() throws Exception {
+		this.delayHandler.setDelayExpression(new SpelExpressionParser().parseExpression("payload"));
+		this.delayHandler.setOutputChannel(new DirectChannel());
+		this.delayHandler.setIgnoreExpressionFailures(false);
+		startDelayerHandler();
+		Calendar releaseDate = Calendar.getInstance();
+		releaseDate.add(Calendar.HOUR, 1);
+		this.delayHandler.handleMessage(new GenericMessage<>(releaseDate.getTime()));
+
+		// emulate restart
+		this.taskScheduler.destroy();
+		MessageGroupStore messageStore = TestUtils.getPropertyValue(this.delayHandler, "messageStore",
+				MessageGroupStore.class);
+		MessageGroup messageGroup = messageStore.getMessageGroup(DELAYER_MESSAGE_GROUP_ID);
+		Message<?> messageInStore = messageGroup.getMessages().iterator().next();
+		Object payload = messageInStore.getPayload();
+		DirectFieldAccessor dfa = new DirectFieldAccessor(payload);
+		long requestTime = (long) dfa.getPropertyValue("requestDate");
+		Calendar requestDate = Calendar.getInstance();
+		requestDate.setTimeInMillis(requestTime);
+		requestDate.add(Calendar.HOUR, -2);
+		dfa.setPropertyValue("requestDate", requestDate.getTimeInMillis());
```

Comment:
???
Revert, please, the changes to the `DelayHandler` and this test will start to fail: there won't be task in the queue, because the message will be released immediately after rescheduling, because the schedule Date is in the past.